### PR TITLE
sdk: add filename and file type for download api

### PR DESCRIPTION
**Pull request by FeelyChau**

### CODE REVIEW

File: packages/sdk/src/base.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export class BaseApi {
   }
 
   /**
-   * trace event
+   * trace event, after `Job.run()`, `Pipeline.install()`, `Plugin.createByName()`, `Plugin.createByTarball()` we can get the `traceId` for tracing the progress event.
```
Comment (yorkie) [inline packages/sdk/src/base.ts:12]: This line is too long :)

File: packages/sdk/src/interface.ts [hunk 2]
```diff
@@ -156,3 +156,10 @@ export interface PipelineConfig {
   name: string;
   plugins: PipelinePlugins;
 }
+
+export type FileDownloadResp = {
+  fileName: string;
```
Comment (yorkie) [inline packages/sdk/src/interface.ts:161]:
```suggestion
  filename: string;
```

File: packages/sdk/src/interface.ts [hunk 3]
```diff
@@ -156,3 +156,10 @@ export interface PipelineConfig {
...
+  fileType: string;
```
Comment (yorkie) [inline packages/sdk/src/interface.ts:162]:
```suggestion
  mimeType: string;
```

File: packages/sdk/src/job.ts [hunk 4]
```diff
@@ -70,8 +70,18 @@ export class Job extends BaseApi {
...
-  downloadOutput(id: string): Promise<FileDownloadResp> {
-    return getFile(`${this.route}/${id}/output`);
+  async downloadOutput(id: string): Promise<FileDownloadResp> {
+    const resp = await getFile(`${this.route}/${id}/output`);
+    // header['content-disposition'] value looks like: 'attachment; filename="pipcook-output-u9fo9dlt.tar.gz"'
+    let fileName = resp.headers['content-disposition']?.match(/\"(.*?)\"/g)[0]?.replace(/\"/g, '');
+    fileName = fileName || 'output.tar.gz';
+    let fileType = resp.headers['content-type'];
+    if (fileType && fileType.startsWith('application/')) {
+      fileType = fileType.substr('application/'.length);
```
Comment (yorkie) [inline packages/sdk/src/job.ts:80]: Because we have used the mime type, so this is no need to call this.

**Review by yorkie — CHANGES_REQUESTED** [review]

File: packages/sdk/src/base.ts (hunk 1 above)
Comment (FeelyChau) [reply packages/sdk/src/base.ts]: Updated.

File: packages/sdk/src/interface.ts (hunk 2 above)
Comment (FeelyChau) [reply packages/sdk/src/interface.ts]: Updated.

File: packages/sdk/src/interface.ts (hunk 3 above)
Comment (FeelyChau) [reply packages/sdk/src/interface.ts]: Updated.

File: packages/sdk/src/job.ts (hunk 4 above)
Comment (FeelyChau) [reply packages/sdk/src/job.ts]: Updated.

File: packages/sdk/src/job.ts [hunk 5]
```diff
@@ -70,8 +70,13 @@ export class Job extends BaseApi {
    * you should check the job status before downloading
    * @param id job id
    */
-  downloadOutput(id: string): Promise<FileDownloadResp> {
-    return getFile(`${this.route}/${id}/output`);
+  async downloadOutput(id: string): Promise<FileDownloadResp> {
+    const resp = await getFile(`${this.route}/${id}/output`);
+    // header['content-disposition'] value looks like: 'attachment; filename="pipcook-output-u9fo9dlt.tar.gz"'
+    let filename = resp.headers['content-disposition']?.match(/\"(.*?)\"/g)[0]?.replace(/\"/g, '');
```
Comment (yorkie) [inline packages/sdk/src/job.ts:76]:
```suggestion
    const filename = resp.headers['content-disposition']?.match(/\"(.*?)\"/g)[0]?.replace(/\"/g, '') || 'output.tgz';
```

Comment (yorkie) [reply packages/sdk/src/job.ts:76]: How about using the library: https://github.com/jshttp/content-disposition?

Comment (FeelyChau) [reply packages/sdk/src/job.ts]: It looks great!

File: packages/sdk/src/job.ts [hunk 6]
```diff
@@ -70,8 +70,13 @@ export class Job extends BaseApi {
    * you should check the job status before downloading
    * @param id job id
    */
-  downloadOutput(id: string): Promise<FileDownloadResp> {
-    return getFile(`${this.route}/${id}/output`);
+  async downloadOutput(id: string): Promise<FileDownloadResp> {
+    const resp = await getFile(`${this.route}/${id}/output`);
+    // header['content-disposition'] value looks like: 'attachment; filename="pipcook-output-u9fo9dlt.tar.gz"'
+    let filename = ContentDisposition.parse(resp.headers['content-disposition']).parameters['filename'];
+    filename = filename || 'output.tar.gz';
```
Comment (yorkie) [inline packages/sdk/src/job.ts:77]: How about merge the above 2 expressions?

Comment (FeelyChau) [reply packages/sdk/src/job.ts]: Updated.

File: packages/sdk/src/job.ts [hunk 7]
```diff
@@ -70,8 +70,12 @@ export class Job extends BaseApi {
    * you should check the job status before downloading
    * @param id job id
    */
-  downloadOutput(id: string): Promise<FileDownloadResp> {
-    return getFile(`${this.route}/${id}/output`);
+  async downloadOutput(id: string): Promise<FileDownloadResp> {
+    const resp = await getFile(`${this.route}/${id}/output`);
+    // header['content-disposition'] value looks like: 'attachment; filename="pipcook-output-u9fo9dlt.tar.gz"'
+    let filename = ContentDisposition.parse(resp.headers['content-disposition']).parameters['filename'] || 'output.tar.gz';
```
Comment (yorkie) [inline packages/sdk/src/job.ts:76]:
```suggestion
    const filename = ContentDisposition.parse(resp.headers['content-disposition']).parameters['filename'] || 'output.tar.gz';
```

File: packages/sdk/src/job.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
-import { get, getFile, del, post, FileDownloadResp } from './request';
+import { get, getFile, del, post } from './request';
 import { BaseApi } from './base';
-import { JobResp, TraceResp, JobListFilter } from './interface';
-
+import { JobResp, TraceResp, JobListFilter, FileDownloadResp } from './interface';
+import * as ContentDisposition from 'content-disposition';
```
Comment (yorkie) [inline packages/sdk/src/job.ts:4]: Shall we directly import `parse` from 'content-disposition'?

Comment (FeelyChau) [reply packages/sdk/src/job.ts]: Yep, it's better.

File: packages/sdk/src/job.ts (hunk 7 above)
Comment (FeelyChau) [reply packages/sdk/src/job.ts]: Fixed.

**Review by yorkie — APPROVED** [review]

**Pull request merged**